subjects/devops/serverless/README.md

```diff
@@ -0,0 +1,159 @@
+
+**Note** Don't forget to add Lambda handler for the application
+
+**2. Use [Cloudformation](https://aws.amazon.com/cloudformation/) to host the lambda function**
+
+Cloudformation is Infrastructure As Code (IaC) tool used to automate the provisioning of AWS resources inside the AWS. It allows you to use a simple yaml syntax to define the resources you want to create.
+
+**2.1 Hosting the Lambda function**
+
+Write cloudformation template with parameters (variables) `SlackWebhookUrl`, `Amount`, `IbanNumber`. The template should automatically provision the Slackbot based on architecture mentioned in [Level 0](#level-0-basic-lambda-function-hosted-via-serverless) step 3.
+
+## Level 2: Basic Lambda Function automation via Basic Terraform with Local State\*\*
+
+Terraform is an Infrastructure As Code (IaC) tool used to automate the provisioning of resources inside different SaaS solutions and Cloud providers. It uses HCL syntax, which is easy maintainable. The main problem of terraform is that you need to know, how everything works under the hood (e.g. IAM roles, policies, serverless platforms etc).
+
+**1. Repeat the 1st and 2nd step from Level 0**
+
+Create a slack bot, that sends a reminder to slack channel. Write integration tests for the application to ensure that each part of the application is functioning properly.
+
+**Note** Don't forget to add Lambda handler for the application
+
+**2. Automate the hosting via Terraform**
+
+**2.1 Write an IAM role with the least privilege principle for the lambda function and EventBridge.**
+
+**2.2 Automatically pack up the lambda function into a zip file**
+
+**2.3 Build the lambda function based on the zip file**
+
+**2.4 Build AWS Eventbridge, which will trigger the lambda function**
+
+**Note** Lambda function should be triggered every month on the 1st day of the month at 10:00 AM.
+
+At this level you can use local state for terraform, no modules needed to write.
+
+## Level 3: Basic Lambda Function automation via Terraform Module with Remote S3 State (advanced)\*\*
+
+In a DevOps world, mostly we use remote state for terraform. The main reason is that we can share the state with other team members and we can easily manage the state of the infrastructure. Although it is a bit more complex to set up, it allows you to avoid issues when someone even removes the state file, cause S3 has versioning turned on.
+
+This level is more advanced, but more near to the real-world scenario. As a DevOps Engineer / SRE mostly all the terraform code is written in modules (or using community pre-built modules). Modules help us to maintain the codebase and reuse the code in different projects, so you don't need to write the same code again and again.
+
+**1. Repeat the steps from [Level 2](#level-2-basic-lambda-function-automation-via-basic-terraform-with-local-state)**
```

Comment:
The link to the section is not correct.
#level-2-basic-lambda-function-automation-via-basic-terraform-with-local-state : 404